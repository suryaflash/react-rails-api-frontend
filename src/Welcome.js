import React, { Component } from 'react';
import './App.css';
import { NavLink } from "react-router-dom";
import { Button, Label, Input ,Alert} from 'reactstrap';
import axios from 'axios';



class Welcome extends Component {


    constructor(props) {
        super(props);
        this.state =
            {
                email: '',
                password: ''
            }
    }

    componentWillMount = () =>
    {
        console.log(localStorage.getItem('email'));
        if(localStorage.getItem('email') !== null )
            window.location.href = "/Add";
    }
    handleChange =(e) =>
    {
        this.setState({[e.target.name] : e.target.value});

    }

    onSignIn = (e) =>
    {
        let userData =
        {
            "auth" :{
            "email" : this.state.email,
            "password" : this.state.password
            }
        }
        axios.post('http://localhost:4000/user/sign_in',userData)
        .then((response) => 
        {
            console.log(response);
            if(response.data.jwt !== null )
            {
                localStorage.setItem('email', this.state.email);
                localStorage.setItem('jwt',response.data.jwt);
                window.location.href = "/Add";    
            }
            else
                window.alert("Invalid Credentials");
        })
        .catch(function (error) {
            console.log(error)
        })
    }

    render() {
        return (
            <div className="container">
                <Alert color="info"> Sign In</Alert>
                <div>

                    <Label >Email</Label>
                    <Input value={this.state.email} id='email' name='email' onChange={this.handleChange} />
                    
                </div>

                <div>
                    <Label >Password</Label>
                    <Input value={this.state.password} type='password' name='password' onChange={this.handleChange}/>
                   
                </div>
                <br/>
                <Button color="success" onClick={this.onSignIn}>SIGN IN</Button>

                <br/>
                <br/>

                Not a User?<NavLink to="/Signup"> Create an Account</NavLink>

            </div>

        )
    }
}

export default Welcome;
