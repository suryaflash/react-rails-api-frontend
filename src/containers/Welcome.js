import React, { Component } from 'react';
import './../App.css';
import { NavLink } from "react-router-dom";
import { Button, Label, Input, Alert } from 'reactstrap';

import { connect } from "react-redux";



class Welcome extends Component {


    constructor(props) {
        super(props);
        this.state =
            {
                email: '',
                password: ''
            }
    }

    componentWillMount = () => {
        console.log(localStorage.getItem('email'));
        if (localStorage.getItem('email') !== null)
            window.location.href = "/articles/new";
    }
    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }


    onSignIn = (e) => {
        let userData =
        {
            "auth": {
                "email": this.state.email,
                "password": this.state.password
            }
        }
        this.props.dispatch({ type: 'signIn', payload: userData })
    }


    render() {
        return (
            <div className="container">
                <Alert color="info"> Sign In</Alert>
                <div>

                    <Label >Email</Label>
                    <Input value={this.state.email} id='email' name='email' onChange={this.handleChange} />

                </div>

                <div>
                    <Label >Password</Label>
                    <Input value={this.state.password} type='password' name='password' onChange={this.handleChange} />

                </div>
                <br />
                <Button color="success" onClick={this.onSignIn}>SIGN IN</Button>

                <br />
                <br />

                Not a User?<NavLink to="/Signup"> Create an Account</NavLink>>

            </div>

        )
    }
}

function mapStateToProps(state) {
    console.log('stp', state);
    return {
        //email : state.email
    }
}

export default connect(mapStateToProps)(Welcome);
