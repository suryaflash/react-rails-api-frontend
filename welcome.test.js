import Enzyme, { shallow } from 'enzyme';
import Welcome from '././src/containers/Welcome.js'
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from 'reactstrap';

Enzyme.configure({ adapter: new Adapter() });

describe('Test case for testing login', () => {
    let wrapper;

    test('username check', () => {
        wrapper = shallow(<Welcome />);
        const t1 = wrapper.find('Input[name="email"]');
        expect(t1).toHaveLength(1);
        t1.simulate('change', { target: { name: 'email', value: 'surya' } });
        expect(wrapper.state('email')).toEqual('surya');
    })

    it('password check', () => {
        wrapper = shallow(<Welcome />);
        const t2 = wrapper.find('Input[name="password"]');
        expect(t2).toHaveLength(1);
        t2.simulate('change', { target: { name: 'password', value: 'surya' } });
        // wrapper.find('Input[type="password"]').simulate('change', {target: {name: 'password', value: 'surya'}});
        expect(wrapper.state('password')).toEqual('surya');
    })

    it('login check with right data', () => {
        wrapper = shallow(<Welcome />);
        const a1 = wrapper.find('Input[name="email"]');
        expect(a1).toHaveLength(1);
        a1.simulate('change', { target: { name: 'email', value: 'surya' } });
        const a2 = wrapper.find('Input[name="password"]');
        expect(a2).toHaveLength(1);
        a2.simulate('change', { target: { name: 'password', value: 'surya' } });
        expect(wrapper.state('email')).toEqual('surya') && expect(wrapper.state('password')).toEqual('surya');
        // wrapper.find('Button').first().simulate('click');
        // wrapper.find('Button').simulate('click');
        expect(true).toBeTruthy();
        expect(a1).toMatchSnapshot();
    })

})