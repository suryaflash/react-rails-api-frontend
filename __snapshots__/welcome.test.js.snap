// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing login login check with right data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Welcome />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            closeAriaLabel="Close"
            color="info"
            fade={true}
            isOpen={true}
            tag="div"
            transition={
              Object {
                "appear": true,
                "baseClass": "fade",
                "baseClassActive": "show",
                "enter": true,
                "exit": true,
                "in": true,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 150,
                "unmountOnExit": true,
              }
            }
          >
             Sign In
          </Alert>,
          <div>
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Email
            </Label>
            <Input
              id="email"
              name="email"
              onChange={[Function]}
              type="text"
              value="surya"
            />
          </div>,
          <div>
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Password
            </Label>
            <Input
              name="password"
              onChange={[Function]}
              type="password"
              value="surya"
            />
          </div>,
          <br />,
          <Button
            color="success"
            onClick={[Function]}
            tag="button"
          >
            SIGN IN
          </Button>,
          <br />,
          <br />,
          "Not a User?",
          <NavLink
            to="/Signup"
          >
             Create an Account
          </NavLink>,
          ">",
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": " Sign In",
            "closeAriaLabel": "Close",
            "color": "info",
            "fade": true,
            "isOpen": true,
            "tag": "div",
            "transition": Object {
              "appear": true,
              "baseClass": "fade",
              "baseClassActive": "show",
              "enter": true,
              "exit": true,
              "in": true,
              "mountOnEnter": false,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 150,
              "unmountOnExit": true,
            },
          },
          "ref": null,
          "rendered": " Sign In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Email
              </Label>,
              <Input
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value="surya"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Email",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Email",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "surya",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Password
              </Label>,
              <Input
                name="password"
                onChange={[Function]}
                type="password"
                value="surya"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Password",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Password",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "surya",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "SIGN IN",
            "color": "success",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "SIGN IN",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Not a User?",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": " Create an Account",
            "to": "/Signup",
          },
          "ref": null,
          "rendered": " Create an Account",
          "type": [Function],
        },
        ">",
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Alert
              closeAriaLabel="Close"
              color="info"
              fade={true}
              isOpen={true}
              tag="div"
              transition={
                Object {
                  "appear": true,
                  "baseClass": "fade",
                  "baseClassActive": "show",
                  "enter": true,
                  "exit": true,
                  "in": true,
                  "mountOnEnter": false,
                  "onEnter": [Function],
                  "onEntered": [Function],
                  "onEntering": [Function],
                  "onExit": [Function],
                  "onExited": [Function],
                  "onExiting": [Function],
                  "tag": "div",
                  "timeout": 150,
                  "unmountOnExit": true,
                }
              }
            >
               Sign In
            </Alert>,
            <div>
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Email
              </Label>
              <Input
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value="surya"
              />
            </div>,
            <div>
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Password
              </Label>
              <Input
                name="password"
                onChange={[Function]}
                type="password"
                value="surya"
              />
            </div>,
            <br />,
            <Button
              color="success"
              onClick={[Function]}
              tag="button"
            >
              SIGN IN
            </Button>,
            <br />,
            <br />,
            "Not a User?",
            <NavLink
              to="/Signup"
            >
               Create an Account
            </NavLink>,
            ">",
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Sign In",
              "closeAriaLabel": "Close",
              "color": "info",
              "fade": true,
              "isOpen": true,
              "tag": "div",
              "transition": Object {
                "appear": true,
                "baseClass": "fade",
                "baseClassActive": "show",
                "enter": true,
                "exit": true,
                "in": true,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 150,
                "unmountOnExit": true,
              },
            },
            "ref": null,
            "rendered": " Sign In",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Label
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  Email
                </Label>,
                <Input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value="surya"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Email",
                  "tag": "label",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": "Email",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "type": "text",
                  "value": "surya",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Label
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  Password
                </Label>,
                <Input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value="surya"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Password",
                  "tag": "label",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": "Password",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "surya",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SIGN IN",
              "color": "success",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": "SIGN IN",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Not a User?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Create an Account",
              "to": "/Signup",
            },
            "ref": null,
            "rendered": " Create an Account",
            "type": [Function],
          },
          ">",
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromError": true,
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "id": "email",
      "name": "email",
      "onChange": [Function],
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "id": "email",
        "name": "email",
        "onChange": [Function],
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            closeAriaLabel="Close"
            color="info"
            fade={true}
            isOpen={true}
            tag="div"
            transition={
              Object {
                "appear": true,
                "baseClass": "fade",
                "baseClassActive": "show",
                "enter": true,
                "exit": true,
                "in": true,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 150,
                "unmountOnExit": true,
              }
            }
          >
             Sign In
          </Alert>,
          <div>
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Email
            </Label>
            <Input
              id="email"
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>,
          <div>
            <Label
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Password
            </Label>
            <Input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>,
          <br />,
          <Button
            color="success"
            onClick={[Function]}
            tag="button"
          >
            SIGN IN
          </Button>,
          <br />,
          <br />,
          "Not a User?",
          <NavLink
            to="/Signup"
          >
             Create an Account
          </NavLink>,
          ">",
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": " Sign In",
            "closeAriaLabel": "Close",
            "color": "info",
            "fade": true,
            "isOpen": true,
            "tag": "div",
            "transition": Object {
              "appear": true,
              "baseClass": "fade",
              "baseClassActive": "show",
              "enter": true,
              "exit": true,
              "in": true,
              "mountOnEnter": false,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 150,
              "unmountOnExit": true,
            },
          },
          "ref": null,
          "rendered": " Sign In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Email
              </Label>,
              <Input
                id="email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Email",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Email",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Password
              </Label>,
              <Input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Password",
                "tag": "label",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": "Password",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "SIGN IN",
            "color": "success",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "SIGN IN",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Not a User?",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": " Create an Account",
            "to": "/Signup",
          },
          "ref": null,
          "rendered": " Create an Account",
          "type": [Function],
        },
        ">",
      ],
      "type": "div",
    },
  ],
}
`;
